pkg/util/utils_test.go

```diff
@@ -57,5 +60,47 @@ func TestRetry(t *testing.T) {
 	if err := Retry(4, f); err == nil {
 		t.Fatalf("Error should have been raised by retry.")
 	}
+}
+
+func errEquals(err1 error, err2 error) bool {
+	if err1 == err2 {
+		return true
+	} else if err1 == nil || err2 == nil {
+		return false
+	} else {
+		return err1.Error() == err2.Error()
+	}
+}
+
+type getLocalkubeArgs struct {
+	input         string
+	expected      string
+	expectedError error
+}
 
+func TestGetLocalkubeDownloadURL(t *testing.T) {
+	inner := func(args getLocalkubeArgs) {
```

Comment:
Sorry for leading you astray by recommending the inner function. I think this might be cleaner if you just inline the function completely into your for loop at the bottom. See this example: https://github.com/golang/go/wiki/TableDrivenTests

But i like keeping the list inside the function instead of at the module level.